cfpb/cfgov-refresh
# Move Event pages rendering logic and templates over to Wagtail control

This is the work that is needed for moving control of Events to Wagtail.

## Additions

- EventFilterForm, EventArchivePage, EventRequestSpeakerPage

## Removals

- Template logic

## Changes

- Django now handles filtering/pagination
- Django/Wagtail is now the backend for the Event pages

## Testing

- Make sure to use `python cfgov/manage.py import-data events eventpage events -u user -p password` to convert the data from Wordpress to Django if you have not done so already
- Visit the /events/ section of the site and make sure the functionality is still there. Basically, is it still the same?

## Review

@rosskarchner 
@kave 
@anselmbradford 
@sebworks 
@KimberlyMunoz 
@jimmynotjim 

## Notes

- Some code was added to be compatible for both django and sheer/elasticsearch backends so functionality wouldn't break in other places. [For example.](https://github.com/cfpb/cfgov-refresh/pull/1080/files#diff-90b0002a8da6f4e8fa486a0aed6d2176R71)

## To do
- [x] Convert templates to be compatible with Django/Wagtail backend
- [x] Move filterable list to Django form
- [x] ~~Move sidenav to view logic~~ Kept because it makes more sense to handle it like this for now.
- [x] Convert macros to `includes` where possible
